### ISSUE TITLE
Rellic should fail gracefully

### ISSUE THREAD

**Issue by pgoodman** [lifting-bits/rellic#210]
Related to #209, Rellic should not contain `CHECK` failures or `LOG(FATAL)` on unsupported language features. Instead, it should have a mechanism to fail gracefully and report why it failed. Lacking a more coherent strategy, throwing an exception and then having the API catch it and convert it in an error value to communicate out of the API could work. 

The key motivator here is a strong desire to use Rellic as a library within a GUI or an interactive system. Right now, we have to conservatively shell out to Rellic as it sometimes tears down the process when a feature or something isn't supported.

**Comment by frabert** [comment]
Just for clarification, `CHECK`s and `LOG(FATAL)` would still be acceptable ways of signaling that "unreachable" states have been detected, right? A sort of explicit invariant checking, that is.

**Comment by pgoodman** [comment]
yes, but these should be for invariant violations that are some deep internal error. If it's something like "we don't support phi nodes" or "we don't support this binary operator" then we should fail gracefully. As much as possible, we want to avoid tearing down the process. A key use case of rellic is in GUI, where we can interactively decompile stuff, and I don't want rellic to tear down the GUI process when we could instead report an error like "Could node decompile for such and such reasons."
